pwndbg/disasm/arch.py

```diff
@@ -503,6 +513,26 @@ def _telescope_format_list(self, addresses: List[int], limit: int, emu: Emulator
                 enhance_string_len=enhance_string_len,
             )
 
+    def _enhance_syscall(self, instruction: PwndbgInstruction, emu: Emulator) -> None:
+        if CS_GRP_INT not in instruction.groups:
+            return None
+
+        syscall_register = pwndbg.lib.abi.ABI.syscall().syscall_register
```

Comment:
Also: if u execute 32-bit syscall via int 0x80, does RAX get truncated to EAX?

Shouldn't we support this here somehow?

Reply:
In terms of caching, it would be safe to do so (for arches that can dynamically change the syscall architecture, we have to handle case-by-case anyways). That being said, this line gets called only whenever a "syscall" instruction is in the disasm view, which is relatively rare.